### PR TITLE
Remove choreo configurations from SMS OTP authenticator in connections page

### PR DESCRIPTION
**Pull request by amanda-ariyaratne**
### Purpose

$subject

### Related Issues
- https://github.com/wso2/product-is/issues/17376

### CODE REVIEW

File: apps/console/src/public/deployment.config.json [hunk 1]
```diff
@@ -472,7 +472,7 @@
                 }
             },
             "smsProviders": {
-                "disabledFeatures": [],
+                "disabledFeatures": ["choreoAsSMSProvider"],
```
Comment (NipuniBhagya) [inline apps/console/src/public/deployment.config.json:475]:
```suggestion
                "disabledFeatures": [ "choreoAsSMSProvider" ],
```

File: apps/console/src/extensions/components/authenticators/sms-otp/sms-otp-authenticator.tsx [hunk 2]
```diff
@@ -16,13 +16,15 @@
  * under the License.
  */
 
+import { AppState } from "apps/console/src/features/core/store";
```
Comment (NipuniBhagya) [inline apps/console/src/extensions/components/authenticators/sms-otp/sms-otp-authenticator.tsx:19]: Shall we please add the relative path here?

File: apps/console/src/extensions/components/authenticators/sms-otp/sms-otp-authenticator.tsx [hunk 3]
```diff
@@ -16,13 +16,15 @@
...
 import {
     CommonAuthenticatorFormInitialValuesInterface,
     CommonAuthenticatorFormMetaInterface
 } from "apps/console/src/features/identity-providers/models/identity-provider";
-import { IdentifiableComponentInterface } from "modules/core/src/models";
+import { FeatureAccessConfigInterface, IdentifiableComponentInterface } from "modules/core/src/models";
```
Comment (NipuniBhagya) [inline apps/console/src/extensions/components/authenticators/sms-otp/sms-otp-authenticator.tsx:24]:
```suggestion
import { FeatureAccessConfigInterface, IdentifiableComponentInterface } from "@wso2is/core/models";
```

File: apps/console/src/public/deployment.config.json (hunk 1 above)
Comment (amanda-ariyaratne) [reply apps/console/src/public/deployment.config.json]: Fixed with 32df88f9368664421e57f487e6477045fa17c5d6

File: apps/console/src/extensions/components/authenticators/sms-otp/sms-otp-authenticator.tsx (hunk 2 above)
Comment (amanda-ariyaratne) [reply apps/console/src/extensions/components/authenticators/sms-otp/sms-otp-authenticator.tsx]: Fixed with 32df88f9368664421e57f487e6477045fa17c5d6

File: apps/console/src/extensions/components/authenticators/sms-otp/sms-otp-authenticator.tsx (hunk 3 above)
Comment (amanda-ariyaratne) [reply apps/console/src/extensions/components/authenticators/sms-otp/sms-otp-authenticator.tsx]: Fixed with 32df88f9368664421e57f487e6477045fa17c5d6

**Review by NipuniBhagya — APPROVED** [review]

**Review by JayaShakthi97 — APPROVED** [review]

**Comment by JayaShakthi97** [comment]
Merging since all the checks passed for the previous commit.

File: apps/console/src/extensions/components/authenticators/sms-otp/sms-otp-authenticator.tsx [hunk 4]
```diff
@@ -78,14 +80,32 @@ export const SmsOTPAuthenticator: FunctionComponent<SmsOTPAuthenticatorInterface
...
 
-    const [ isReadOnly, setIsReadOnly ] = useState<boolean>(true);
+    const featureConfig: FeatureAccessConfigInterface = useSelector((state: AppState) => {
+        return state.config?.ui?.features?.smsProviders;
+    });
+
+    const isChoreoEnabledAsSMSProvider: boolean = useMemo(() => {
+        const disabledFeatures: string[] = featureConfig?.disabledFeatures;
+
+        return !disabledFeatures.includes("choreoAsSMSProvider");
```
Comment (JayaShakthi97) [inline apps/console/src/extensions/components/authenticators/sms-otp/sms-otp-authenticator.tsx:90]: Null check needs to be added here. If the feature config is not defined in the deployment config, `disabledFeatures` will be `undefined`

**Pull request merged**